src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -2331,6 +2340,17 @@ internal IAsyncResult BeginConnect(string host, int port, AsyncCallback requestC
                 throw new InvalidOperationException(SR.net_sockets_mustnotlisten);
             }
 
+            IPAddress parsedAddress;
```

Comment:
Same as above: this extra processing is required only for Unix and so should be part of the respective PAL layer.